src/oph/heratepalvelu/amis/AMISCommon.clj

```diff
@@ -44,11 +77,16 @@
                           (:koodiarvo (:suorituskieli suoritus)))
           rahoitusryhma (c/get-rahoitusryhma opiskeluoikeus
                                              herate-date)]
-      (if (c/check-duplicate-herate? oppija
-                                     koulutustoimija
-                                     laskentakausi
-                                     kyselytyyppi
-                                     herate-source)
+      (if-some [existing (c/already-superseding-herate! oppija
+                                                        koulutustoimija
+                                                        laskentakausi
+                                                        kyselytyyppi
+                                                        herate-source)]
+        (log/info "Meillä on jo vastaava heräte" existing
+                  "joten ei ylikirjoiteta. Uudessa oppija:" oppija
+                  "koulutustoimija:" koulutustoimija
+                  ";" kyselytyyppi laskentakausi)
+
```

Comment:
Tähän funktioon tehdyissä muutoksissa minulla ei sinänsä mitään huomautettavaa. Funktio on melko pitkä, ja se on tämän refaktoroinnin seurauksena onneksi lyhentynyt jonkin verran. Mietin, että olisiko `db-data`n luominen (mukaanlukien `req-body` ja `db-data-cond-values`, rivit 53-97 vanhassa koodissa) hyvä eriyttää omaan funktioonsa, esim. `build-db-data` tjmv. Voisi ottaa parametrina samat kuin `check-and-save-herate!`, jolloin ylempänä tehtävien let-bindingien määräkin vähenisi hieman.

Reply:
tästä vois ehkä tehdä vielä oman tikettinsä.  Tiedän etten oo itsekään kunnolla torjunut kiusausta tehdä refaktorointeja, mutta raja on vedettävä johonkin (herätepalvelussa kun on aika paljon refaktoroitavaa).

Reply:
(Oon siis ilman muuta samaa mieltä että kyseistä funktiota on syytä lyhentää mm. juuri siksi, että datadependenssit olisivat helpommin seurattavissa.)